---
title: "CMap Prep"
author: "Jeff Brabec"
date: "2/14/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(here)
library(janitor)
library(safejoin)
library(gprofiler2)
library(yardstick)
library(viridis)
library(readxl)
library(ggrepel)
library(safejoin)
source(here("src/get_deg_aucs.R"))
source(here("src/convert_to_names.R"))
source(here("src/final_auc_processing.R"))
source(here("src/gene_rank_plotr.R"))
source(here("src/get_plot_cs.R"))
source(here("src/make_pan_cell_sets.R"))
source(here("src/get_fpr.R"))
```

## Read in Scores

```{r}
#Linear Results
linear_astro_final_scores <- read_csv(here("results/final_scores/linear_astro_final_scores.csv"))
linear_glia_final_scores <- read_csv(here("results/final_scores/linear_glia_final_scores.csv"))
linear_neuron_final_scores <- read_csv(here("results/final_scores/linear_neuron_final_scores.csv"))
linear_mn_final_scores <- read_csv(here("results/final_scores/linear_mn_final_scores.csv"))
```

## Calculate FPR and TPR for Genes
```{r}
linear_astro_fpr <- get_fpr(linear_astro_final_scores)
linear_glia_fpr <- get_fpr(linear_glia_final_scores)
linear_neuron_fpr <- get_fpr(linear_neuron_final_scores)
linear_mn_fpr <- get_fpr(linear_mn_final_scores)
```

## No Path v. Early Path

### Read in scRNAseq Data

Here we use the custom function `get_scrnaseq_data` which reads in a specified section of the excel sheet and splits that data into UP, DOWN, and ALL genes based on Z score.

```{r}
#Excitatory Neuron Data
ex_neu_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/ex_neu_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)

#Inhibitory Neuron Data
in_neu_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/in_neu_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)

#Astrocyte Data
ast_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/ast_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)

#Microglia Data
micro_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/micro_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)

#Oligodendrocyte Data
olig_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/olig_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)

#Oligodendrocyte Precursor Cell Data
opc_np_ep_degs <- read_rds(here("data/single_cell_data/no_path_v_early_path/opc_np_ep_degs.rds")) %>% map(., mutate, model_p = mixed_model_p)
```

## Get DEGS For excitatory Neuron
```{r}
ex_mn_genes_up <- ex_neu_np_ep_degs$UP %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05)) %>% 
  arrange(desc(log_p)) %>% 
  slice_head(n = 150) %>% 
  select(gene)

write_delim(ex_mn_genes_up, here("results/cmap/ex_neu_up_tags.txt"), delim = "\t", col_names = FALSE)

ex_mn_genes_down <- ex_neu_np_ep_degs$DOWN %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05)) %>% 
  arrange(desc(log_p)) %>% view()
  slice_head(n = 150) %>%
  select(gene)

write_delim(ex_mn_genes_down, here("results/cmap/ex_neu_down_tags.txt"), delim = "\t", col_names = FALSE)


ex_mn_genes_up_func <- ex_neu_np_ep_degs$UP %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  arrange(desc(log_p)) %>% 
  slice_head(n = 150) %>% 
  select(gene)

write_delim(ex_mn_genes_up_func, 
            here("results/cmap/ex_neu_up_func_tags.txt"), 
            delim = "\t", 
            col_names = FALSE)

ex_mn_genes_down_func <- ex_neu_np_ep_degs$DOWN %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  arrange(desc(log_p)) %>% 
  slice_head(n = 150) %>% 
  select(gene)

write_delim(ex_mn_genes_down_func, 
            here("results/cmap/ex_neu_down_func_tags.txt"), 
            delim = "\t", 
            col_names = FALSE)



filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "Ex. Neuron")
```

## Combine Multi-net Scores with DEG data
```{r}
ex_mn_genes <- ex_neu_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>%
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "Ex. Neuron")

in_mn_genes <- in_neu_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% filter(log_p > -log10(0.05))
filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "In. Neuron")

ast_mn_genes <- ast_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "Astrocyte")

micro_mn_genes <- micro_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "Microglia")

olig_mn_genes <- olig_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "Oligodendrocyte")

opc_mn_genes <- opc_np_ep_degs$ALL %>% 
  inner_join(linear_mn_fpr, by = c("gene" = "gene_name")) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  filter(log_p > -log10(0.05), log_fpr > 1) %>% 
  select(gene, log_fpr, log_p) %>% 
  mutate(cell_type = "OPC")
```

```{r}
ex_neu_genes <- read_xlsx(here("data/single_cell_data/41586_2019_1195_MOESM4_ESM.xlsx"), 
                          sheet = "Ex", 
                          range = "L2:T16846")

neuron_scores <- ex_neu_genes %>% 
  janitor::clean_names() %>% 
  rename(gene_name = x1) %>% 
  left_join(linear_mn_fpr, by = "gene_name") %>% 
  filter(!is.na(class)) %>% 
  select(gene_name, log_fpr, 
         mixed_model_p, 
         ind_model_adj_pvals, 
         ind_model_fc, 
         mixed_model_z,
         de_gs_ind_model, 
         de_gs_ind_mix_models)  %>% 
  mutate(mixed_model_p = replace_na(mixed_model_p, 1)) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  rowwise() %>% 
  mutate(comb_score = sum(log_fpr > .$log_fpr & log_p > .$log_p)) %>% 
  ungroup() %>% 
  mutate(comb_score = comb_score/n()) %>% 
  arrange(desc(comb_score)) 

neuron_scores %>%
  filter(mixed_model_z > 0) %>% 
  select(gene_name) %>% 
  slice_head(n = 150) %>% 
  write_delim(here("results/cmap/comb_score_top.txt"), delim = "/t", col_names = FALSE)

neuron_scores %>% 
  filter(mixed_model_z < 0) %>% 
  select(gene_name) %>% 
  slice_head(n = 150) %>% 
  write_delim(here("results/cmap/comb_score_bottom.txt"), delim = "/t", col_names = FALSE)
```

## For Late shenanigans
```{r}
late_ex_neu_genes <- read_xlsx(here("data/single_cell_data/41586_2019_1195_MOESM4_ESM.xlsx"), 
                          sheet = "Ex", 
                          range = "W2:AE16444")

late_neuron_scores <- late_ex_neu_genes %>% 
  janitor::clean_names() %>% 
  rename(gene_name = x1) %>% 
  left_join(linear_mn_fpr, by = "gene_name") %>% 
  filter(!is.na(class)) %>% 
  select(gene_name, log_fpr, 
         mixed_model_p, 
         ind_model_adj_pvals, 
         ind_model_fc, 
         mixed_model_z,
         de_gs_ind_model, 
         de_gs_ind_mix_models)  %>% 
  mutate(mixed_model_p = replace_na(mixed_model_p, 1)) %>% 
  mutate(log_p = -log10(mixed_model_p)) %>% 
  rowwise() %>% 
  mutate(comb_score = sum(log_fpr > .$log_fpr & log_p > .$log_p)) %>% 
  ungroup() %>% 
  mutate(comb_score = comb_score/n()) %>% 
  arrange(desc(comb_score)) 

late_neuron_scores %>%
  filter(mixed_model_z > 0) %>% 
  select(gene_name) %>% 
  slice_head(n = 150) %>% 
  write_delim(here("results/cmap/late_comb_score_top.txt"), delim = "/t", col_names = FALSE)

late_neuron_scores %>% 
  filter(mixed_model_z < 0) %>% 
  select(gene_name) %>% 
  slice_head(n = 150) %>% 
  write_delim(here("results/cmap/late_comb_score_bottom.txt"), delim = "/t", col_names = FALSE)
```

